lib/spack/spack/cmd/list.py

```diff
@@ -157,7 +161,8 @@ def rst(pkg_names):
     print('automatically generated based on the packages in the latest Spack')
     print('release.')
     print()
-    print('Spack currently has %d mainline packages:' % len(pkgs))
+    print('Spack currently has {0} mainline packages matching {1}:'
+          ''.format(len(pkgs), ' '.join(query)))
```

Comment:
All these query changes should only occur if query is non-empty. The rst and html formatters are used to generate https://spack.readthedocs.io/en/latest/package_list.html, where "mainline packages matching :" doesn't make a lot of sense.

Reply:
I like having them the same, but this is also important for name_only. If I run `spack list` I don't want to see an empty query.

Reply:
Do you have less ugly suggestion on how to do this? 
```
'somestring' + 'someotherstring' if False else 'bla'
```
evaluates to `'bla'`.... hence I needed to introduce the `msg` thingy...